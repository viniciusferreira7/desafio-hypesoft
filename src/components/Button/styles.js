import styled, { css } from 'styled-components';

export const Container = styled.button`
${({ theme, background, sizes }) => css`
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  padding: 1.2rem ${theme.spacings.large};
  letter-spacing: 0.02rem;
  text-transform: uppercase;
  background:${background ? theme.colors.secondaryColor : theme.colors.white};
  color:${background ? theme.colors.white : theme.colors.primaryColor};
  font-size: ${sizes}rem;
  border-radius: 5px;
  border: none;
  transition: all 500ms ease-in-out;
  cursor: pointer;

&:hover {
  background:${!background ? theme.colors.secondaryColor : theme.colors.white};
  color:${!background ? theme.colors.white : theme.colors.primaryColor};
}
`}
`;
