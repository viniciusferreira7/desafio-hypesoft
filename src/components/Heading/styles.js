import styled, { css } from 'styled-components';

const titleSize = {
  small: (theme) =>
    css`
      font-size: ${theme.font.sizes.large};
    `,
  medium: (theme) =>
    css`
      font-size: ${theme.font.sizes.xlarge};
    `,
  big: (theme) =>
    css`
      font-size: ${theme.font.sizes.xxlarge};
    `,
  huge: (theme) =>
    css`
      font-size: ${theme.font.sizes.xhuge};
      ${mediaFont(theme)}
    `,
};

const mediaFont = (theme) => css`
  @media ${theme.media.lteMedium} {
    font-size: ${theme.font.sizes.xlarge};
  }
`;

export const Title = styled.h1`
  ${({ theme, size }) => css`
    color: ${theme.colors.white};
    ${titleSize[size](theme)};
  `}
`;
