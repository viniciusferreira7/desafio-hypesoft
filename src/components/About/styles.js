import styled, { css } from 'styled-components';
import { Container as Button } from '../Button/styles';
import { Container as TextContainer } from '../TextComponent/styles';

export const Container = styled.div`
${() => css`
  width: 100%;
  display: grid;
  grid-template-columns: 1fr 1fr;
  grid-column-gap: 2.8rem;
  grid-template-areas:'SrcImg' 'Description';
  padding-top: 10rem;
`}
`;

export const SrcImg = styled.div`
${({ theme, srcImg }) => css`
  grid-area: 'SrcImg';
  max-width: 56rem;
  width: 100%;
  height: 54.2rem;
  background-image: url(${srcImg});
  background-size: cover;
  border-radius: 1rem;
  box-shadow: 3.2rem 3.2rem   ${theme.colors.white};

  @media ${theme.media.lteMedium}{
    display: none;
  }
`}
`;

export const Description = styled.div`
${() => css`
  grid-area: 'Description';
  max-width: 67.2rem;

  > ${Button} {
    width: 16.2rem;
    height: 5.6rem;

    > img{
      margin-left: 1.1rem;
    }
  }
`}
`;

export const InfoHeading = styled.div`
${() => css`
`}
`;

export const InfoContainer = styled.div`
${({ theme }) => css`
  max-width: 67.2rem;
  display: flex;

  @media ${theme.media.lteMedium}{
    flex-direction: column;
    padding-bottom: 3.2rem;
  }
`}
`;

export const Group = styled.div`
${() => css`
  max-width: 20.6rem;
  display: flex;
  align-items: center;
  margin-right: 5rem;

  > ${TextContainer}:nth-child(2){
    max-width: 9.5rem;
    margin-left: 1.6rem;
    padding: 0;
  }
`}
`;

`}`;

export const GroupTwo = styled.div`
${() => css`
  width: 25.1rem;
  display: flex;
  align-items: center;

  > ${TextContainer}:nth-child(2){
    max-width: 9.5rem;
    margin-left: 1.6rem;
    padding: 0;
  }
`}
`;

`}`;
