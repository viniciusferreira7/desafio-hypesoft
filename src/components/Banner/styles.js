import styled, { css } from 'styled-components';
import { Title as Heading } from '../Heading/styles';
import { Container as Button } from '../Button/styles';
import { Container as GridDot } from '../GridDot/styles';

export const Container = styled.section`
${({ theme }) => css`
  width: 100%;
  max-height:31.2rem;
  overflow: hidden;
  padding: 6.4rem 2%;
  position: relative;
  margin-top: 16.4rem;

  @media ${theme.media.lteMedium}{
      max-height:auto;
      padding-top: 3rem
    }

 >  ${GridDot}{
    opacity: 0.15;
    position: absolute;
    top: 6.4rem;
    left: 12.2rem;
  }


  ${Heading}{
      width:50rem;
      height: 9.6rem;
      margin: 0 auto;


      @media ${theme.media.lteMedium}{
        width: 100%;
        text-align: center;
        height: 15rem;
        margin-bottom: 7rem;
      }

      > span{
        color:#bc199c;
      }

      > span::before{
        display: none;
      }
    }

  > ${Button} {
      width: 17.7rem;
      height: 5.6rem;
      margin: 0 auto;
      margin-top: 7.2rem;

      > a{
        margin-right: 0.9rem;
      }
  }
`}
`;

export const SmallEllipse = styled.div`
  ${({ theme }) => css`
    width:7.6rem;
    height:7.6rem;
    border-radius: calc(7.6rem / 2);
    background: ${theme.colors.white};
    opacity: 0.03;
    position: absolute;
    top: 2.6rem;
    right: 11.6rem;

    @media ${theme.media.lteMedium}{
      top: 8rem;
      right: 0;
    }
  `}
`;

export const HugeEllipse = styled.div`
  ${({ theme }) => css`
    width:52.9rem;
    height:52.9rem;
    border-radius: calc(52.9rem / 2);
    background: ${theme.colors.white};
    opacity: 0.03;
    position: absolute;
    top: 13.9rem;
    right: 90.7rem;

    @media ${theme.media.lteMedium}{
      width:30rem;
      height:30rem;
      top: 0;
      right: 4rem;
    }
  `}
`;

export const ContainerGridDot = styled.div`
  ${() => css`
    position: absolute;
    top:15rem;
    right: 21rem;
    transform: rotate(-90deg);

    ${GridDot}{
      opacity: 0.3;

    }
  `}
`;
