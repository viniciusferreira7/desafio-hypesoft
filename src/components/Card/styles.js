import styled, { css } from 'styled-components';
import { Container as TextContainer } from '../TextComponent/styles';

export const Container = styled.div`
${({ theme }) => css`
  background: linear-gradient(0deg, rgba(223, 227, 235, 0.4), rgba(223, 227, 235, 0.4)), #FFFFFF;
  border-radius: 6px;
  width: 31.2rem;
  max-height: 35.1rem;
  transition: all 400ms ease-in-out;

  @media ${theme.media.lteMedium}{
    width: 18.5rem;
    padding-bottom: 2%;
    max-height: 40rem;

    > svg {
      width:2.5rem !important;
      height:2.5rem !important;
      margin: 0;
    }

    ${TextContainer}{
      text-align: center;
      font-size: 2rem;
      padding:2.2rem 0 3.2rem 0 !important ;

      > strong {
        margin-top: 1.6rem;
      }
    }

  }

    > svg{
      width:4rem;
      height:4rem;
      margin: 2.2rem 2.2rem 0 2.2rem;
      color: #242424;
    }

    ${TextContainer}{
      padding:2.2rem 0 3.2rem 2.8rem ;
      position: relative;

      > strong {
        width: 6.4rem;
        height: 0.3rem;
        background-color: #ED1CA6;
        margin-top: 1.6rem;
        margin-bottom: 2.8rem;
        position: absolute;
        bottom: -1.3rem;
        left:6rem ;
      }
    }

    > img{
      margin: 2.8rem 0 3.6rem 3.5rem;
    }


    &:hover{
      background: #FFFFFF;
      box-shadow: 0px 20px 48px rgba(10, 25, 49, 0.09);
      border-radius: 6px;
      border: 0.3rem solid #ED1CA6;

      ${TextContainer}{
        color:#ED1CA6 ;
      }

      > svg {
        color: #ED1CA6;
      }
    }

;
`}
`;
