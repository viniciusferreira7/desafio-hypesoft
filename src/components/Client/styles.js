import styled, { css } from 'styled-components';
import { Container as TextComponent } from '../TextComponent/styles';

export const Container = styled.div`
${() => css`
  display: flex;
  align-items: center;
  width: 31.2rem;
  height: 10rem;
  background: linear-gradient(0deg, rgba(223, 227, 235, 0.3), rgba(223, 227, 235, 0.3)), #FFFFFF;
  border-radius: 0.4rem;
`}
`;

export const Photo = styled.div`
${({ theme, srcImg }) => css`
  width: 5.2rem;
  height: 5.2rem;
  border-radius: calc(5.2rem / 2);
  background-image:url(${srcImg});
  background-size: cover;
  background-color: ${theme.colors.gray_500};
  margin: 2.4rem;
  `}
`;

export const Info = styled.div`
${() => css`
  ${TextComponent}{
    text-transform: capitalize;
    padding: 0;
    padding-bottom: 0.6rem;

    > span{
      color: #ED1CA6
;
    }
  }
`}
`;
