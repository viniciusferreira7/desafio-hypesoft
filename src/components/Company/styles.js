import styled, { css } from 'styled-components';
import { Title as Heading } from '../Heading/styles';

export const Container = styled.div`
${({ theme }) => css`
  width: 100%;
  display: flex;
  padding-top: 20rem;
  justify-content: space-around;

  @media ${theme.media.lteMedium}{
    flex-direction: column;
    text-align: center;
  }

  ${Heading}{
    width: 312px;

    > strong{
      color:#bc199c;
      font-weight: 400;
    }
  }
`}
`;

export const ContainerLogo = styled.div`
${({ theme }) => css`
  display: flex;
  max-width: 93.6rem;
  max-height: 12.0rem;
  justify-content: space-around;

  @media ${theme.media.lteMedium}{
    flex-direction: column;
    width: 100%;
    max-height: 96.6rem;
  }
`}
`;
export const Logo = styled.div`
${({ theme }) => css`
  display: flex;
  align-items: center;
  justify-content: center;
  width: 168px;
  height: 120px;
  margin-right: 2.4rem;
  background: linear-gradient(0deg, rgba(223, 227, 235, 0.3), rgba(223, 227, 235, 0.3)), #FFFFFF;
  border-radius: 0.8rem;
  transition: all 500ms ease-in-out;

  @media ${theme.media.lteMedium}{
    margin: 0 auto;
    margin-top: 3rem;
  }

  &:hover{
    border:1rem solid ${theme.colors.white};
  }

  >img{
    width: 11.304rem;
    height: 2.6rem;
  }

  >img:nth-of-type(5){
    width:2.756rem !important;
    height:1.205rem !important;
  }

;

`}
`;
