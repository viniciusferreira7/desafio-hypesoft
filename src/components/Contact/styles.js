import styled, { css } from 'styled-components';
import { Container as GridDot } from '../GridDot/styles';

export const Container = styled.section`
${({ theme }) => css`
  width: 100%;
  display: flex;
  padding: 10rem;
  align-items: center;
  margin-top: 22.7rem;
  position: relative;

  ${GridDot}{
    position: absolute;
    top: 16%;
    right:12%;
    transform: rotate(-90deg);
    opacity: 0.3;

    @media ${theme.media.lteMedium}{
      display: none;
  }
  }

  @media ${theme.media.lteMedium}{
    padding: 2rem;
    flex-direction: column;
  }
`}`;
