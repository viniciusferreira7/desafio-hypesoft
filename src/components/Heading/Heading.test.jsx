import { screen } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { Heading } from '.';
import { renderTheme } from '../../styles/render-theme';
import { theme } from '../../styles/theme';

describe('<Heading/>', () => {
  it('should render with default values', () => {
    renderTheme(<Heading>Texto</Heading>);
    const heading = screen.getByRole('heading', { name: 'Texto' });

    expect(heading).toHaveStyle({
      color: theme.colors.white,
    });
  });

  it('should render with correct heading sizes', () => {
    const { rerender } = renderTheme(<Heading size="small">Texto</Heading>);
    const heading = screen.getByRole('heading', { name: 'Texto' });

    expect(heading).toHaveStyle({
      'font-size': theme.font.sizes.medium,
    });

    rerender(
      <ThemeProvider theme={theme}>
        <Heading size="medium">Texto</Heading>
      </ThemeProvider>,
    );

    expect(heading).toHaveStyle({
      'font-size': theme.font.sizes.large,
    });

    rerender(
      <ThemeProvider theme={theme}>
        <Heading size="medium">Texto</Heading>
      </ThemeProvider>,
    );

    expect(heading).toHaveStyle({
      'font-size': theme.font.sizes.large,
    });

    rerender(
      <ThemeProvider theme={theme}>
        <Heading size="large">Texto</Heading>
      </ThemeProvider>,
    );

    expect(heading).toHaveStyle({
      'font-size': theme.font.sizes.xlarge,
    });
    rerender(
      <ThemeProvider theme={theme}>
        <Heading size="big">Texto</Heading>
      </ThemeProvider>,
    );

    expect(heading).toHaveStyle({
      'font-size': theme.font.sizes.xxlarge,
    });

    rerender(
      <ThemeProvider theme={theme}>
        <Heading size="huge">Texto</Heading>
      </ThemeProvider>,
    );

    expect(heading).toHaveStyle({
      'font-size': theme.font.sizes.xhuge,
    });
  });

  it('should render with correct font-size when using mobile', () => {
    renderTheme(<Heading size="huge">Texto</Heading>);
    const heading = screen.getByRole('heading', { name: 'Texto' });

    expect(heading).toHaveStyleRule('font-size', theme.font.sizes.xlarge, {
      media: theme.media.lteMedium,
    });
  });

  it('should render with bold', () => {
    const { rerender } = renderTheme(<Heading bold>Texto</Heading>);
    const heading = screen.getByRole('heading', { name: 'Texto' });

    expect(heading).toHaveStyle({
      'font-weight': '500',
    });

    rerender(
      <ThemeProvider theme={theme}>
        <Heading bold={false}>Texto</Heading>
      </ThemeProvider>,
    );
    expect(heading).toHaveStyle({
      'font-weight': '400',
    });
  });

  it('should render correct heading element', () => {
    const { container } = renderTheme(<Heading as="h6">Texto</Heading>);
    // const heading = screen.getByRole('heading', { name: 'Texto' });

    const h6 = container.querySelector('h6');
    expect(h6.tagName.toLowerCase()).toBe('h6');
  });

  it('should match snapshot', () => {
    const { container } = renderTheme(<Heading as="h6">Texto</Heading>);
    // const heading = screen.getByRole('heading', { name: 'Texto' });

    expect(container).toMatchSnapshot();
  });
});
