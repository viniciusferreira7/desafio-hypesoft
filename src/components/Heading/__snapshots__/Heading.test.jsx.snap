// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Heading/> should match snapshot 1`] = `
.c0 {
  color: #fff;
  font-size: 3.2rem;
  text-transform: none;
}

<div>
  <h6
    class="c0"
  >
    Texto
  </h6>
</div>
`;
